### PR TITLE
Use new modal to confirm Sign Out action - Closes #735

### LINKED ISSUE #735: Use our custom modal to confirm for signing out
### Description
We should use our custom modal instead of the native one, in order to provide more information and improve the user experience by keeping the modal views consistent. 

### Requirements
This requires @reyraa to ensure the designs and translations are in place.

### PR DESCRIPTION
**Pull request by altayaydemir**
# What was the bug or feature?
Described in #735

### How did I fix it?
- Simplified `SignOutButton` component and updated props passed to it in `Settings` page.
- Created `SignOutModal` with the new design and handles the confirmation task.

## Type of change
- Enhancement (a non-breaking change which adds functionality)

### How to test it?
- Login
- Go to Settings Screen
- Try to Sign Out

# Checklist:
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/settings/signOutModal/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      <Small style={[styles.text, styles.theme.text]}>
+        {t('Are you sure you want to sign out?')}
+      </Small>
+    );
+
+    if (settings.hasStoredPassphrase) {
+      content = (
+        <React.Fragment>
+          <Small style={[styles.text, styles.theme.text]}>
+            {t('Signing out will disable bioAuth for the Lisk App.', { sensorType: settings.sensorType })}
```
Comment (slaweet) [inline src/components/settings/signOutModal/index.js]: Why `sensorType` is not used in the message?

```suggestion
            {t('Signing out will disable {{sensorType}} for the Lisk App.', { sensorType: settings.sensorType })}
```

**Comment by aydieneue** [comment]
@yasharAyari I believe this topic is out of the context of this PR and it's related to the initial implementation of this modal component. We are already using this on while switching the active token and we have the same delay there. What I can suggest is to open another issue for fixing that behavior.

**Comment by aydieneue** [comment]
Adding `suspended` label to this one until we come up with a final decision about the modal component.

File: src/components/settings/signOutModal/index.js (hunk 1 above)
Comment (yasharAyari) [reply src/components/settings/signOutModal/index.js]: @slaweet I talked to the design team they needed to have this `X` button inside of the all modals. About the sensor type, it is different on any devices. So, it makes sense to have as a variable in our translation string.

**Pull request merged**